daml-foundations/daml-ghc/ide/src/Development/IDE/State/Rules/Daml.hs

```diff
@@ -177,10 +177,14 @@ contextForFile file = do
     encodedModules <-
         mapM (\m -> fmap (\(hash, bs) -> (hash, (LF.moduleName m, bs))) (encodeModule lfVersion m)) $
         NM.toList $ LF.packageModules pkg
+    DamlEnv{..} <- getDamlServiceEnv
     pure SS.Context
         { ctxModules = Map.fromList encodedModules
         , ctxPackages = map (\(pId, _, p, _) -> (pId, p)) (Map.elems pkgMap)
         , ctxDamlLfVersion = lfVersion
+        , ctxLightValidation = case envScenarioValidation of
+              ScenarioValidationFull -> SS.LightValidation False
```

Comment:
Why not just use the ScenarioValidation type here instead of LightValidation Bool?

Reply:
`ScenarioValidation` is currently in `//daml-foundations/daml-ghc/ghc-compiler` (whether that’s a good name is another question, there is definitely DAML-specific stuff in there) which does not depend on the scenario service client. On the other hand the scenario service client does not depend on `ghc-compiler`. So using the same type would make the dependency tree less parallel which I was trying to avoid. That said, given that the scenario service client doesn’t change too often it would probably be reasonable to make `ghc-compiler` depend on it but we might want to avoid until we finally reorganize the whole mess in `daml-foundations`.

Reply:
Fair enough. Happy to leave where it is for now.